### ISSUE TITLE
URL override used by server (Hub) but not expected by client

### ISSUE THREAD

**Issue by gkustas** [fhircast/sandbox#26]
I am new to this group, and just ran my first tests today. I installed and ran the sandbox code as delivered here. I was unable to post a successful subscription and was able to troubleshoot the problem. I'm hoping (and expecting) it is a configuration issue my side that can be easily explained.

I start both the WebSubClient and Hub using the Visual Studio debugger. From the client, I post a subscription event. I can trace everything working successfully until the hub's SubscriptionValidator issues the callback to the client requesting validation. 

The problem is that the  SubscriptionValidator uses the SubscriptionCallback class method GetCallbackURI which has a URL override inserting "hub." before each URI parameter. For example:

hub.callback=xxxx&hub.challenge=yyy

So, on the client side, the web method is called:

        [HttpGet("{subscriptionId}")]
        public IActionResult Get(string subscriptionId, [FromQuery] SubscriptionVerification verification) {

Of course, the verification can not be bound because of the URL overrides. The client expects :challenge", not "hub.challenge", etc... 

Can someone clarify if this is intentional, and if so, why the client isn't using the same convention?

**Comment by lbergnehr** [comment]
Thanks for the report, @gkustas! This is certainly not intentional and is a recent regression due to another fix. I'll have a look at it as soon as I can. In the meantime, to get something up and running, you could try running without b4ef7ce68b8f2f36933a15dedcd72aa8db5225e1, as that introduces the change.

**Comment by gkustas** [comment]
Thanks @lbergnehr . I went ahead and made local changes to bypass the URLOverride (just commented it out in Subscription.cs model) and changed the code slightly in subscriptioncallback.cs so that it just uses the property name (not the override).

I also noticed a problem here trying to define the events as a string array. The code is always converting back and forth between a coma-delimited string to form and array, but fails to do it in subscriptioncallback.cs. I suggest that the events be defined in the model as a comma delimited string and if/when necessary converted to an array for easier procession in a loop.

**Comment by gkustas** [comment]
After further investigation, I was able to figure out how to keep the "hub" prefix (since it is actually part of the FHIRCast spec). The change required to make this work on the client receiving hook was to use the FromQuery Attribute in the model. Example in Subscriptions.cs in models:

        [BindRequired]
        [FromQuery(Name ="hub.callback")]
	public Uri Callback { get; set; }

Notice also that I removed the implementation of the custom URLOverride attribute. When creating the URI callback for the subscsription verification, I changed the code in SubscriptionCallback.cs to "hard code the "hub." instead of using the custom attribute. I suppose either is OK, but it seems like overkill to use the Attribute when the callback parameters are specific to the FHIRCast verification specification. Code:

    public class SubscriptionCallback {
        public Uri GetCallbackUri(Subscription subscription, SubscriptionBase parameters) {
			var properties = parameters.GetType().GetProperties()
				.Where(x => x.GetValue(parameters, null) != null)
				.Select(x => "hub." + x.Name + "=" + HttpUtility.UrlEncode(x.GetValue(parameters, null).ToString()));
